benches/search_comparison.rs

```diff
@@ -285,3 +240,64 @@ fn make_sorted_vec<T: Ord>(mut v: Vec<T>) -> Vec<T> {
 fn search_sorted_vec<'a, T: Ord>(c: &'a Vec<T>, x: T) -> Option<&'a T> {
     c.binary_search(&x).ok().map(|i| &c[i])
 }
+
+fn test_sequence<T>(duplicates: bool, max: usize) -> impl Iterator<Item = T>
+where
+    T: TryFrom<usize>,
+    <T as TryFrom<usize>>::Error: core::fmt::Debug,
+{
+    let mut index = 0;
+    std::iter::from_fn(move || {
+        let item = if duplicates {
+            let item = std::cmp::min(index, max);
+            item % 256
+        } else {
+            std::cmp::min(index * 2, max)
+        };
+        index += 1;
+
+        Some(T::try_from(item).unwrap())
+    })
+}
+
+/// Generate pseudorandom sequence of numbers
+///
+/// Without duplicates:
+/// ```rust
+/// assert_eq!(
+///     pseudorandom_iter::<u8>(16, 0, false, u8::MAX).take(16).collect(),
+///     vec![30, 4, 18, 8, 6, 12, 26, 16, 14, 20, 2, 24, 22, 28, 10, 0]
+/// )
+/// ```
+///
+/// With duplicates:
+/// ```rust
+/// assert_eq!(
+///     pseudorandom_iter::<u32>(16, 0, true, u32::MAX).take(16).collect(),
+///     vec![15, 14, 9, 8, 5, 10, 0, 4, 5, 0, 4, 6, 5, 14, 12, 2]
```

Comment:
It seems strange to me that all values are such small considering that `max = 2^32`.

Reply:
Yeah because it depends on a `duplicate` flag. If it's `true` we need to bound numbers by the `size` so we could have some duplicates present